code_challenges/advent_of_code/hydrothermal_venture.py

```diff
@@ -0,0 +1,145 @@
+def get_vents_input():
+    with open("code_challenges/advent_of_code/vents_input") as vents_input:
+        vents_coordinates = vents_input.read().split("\n")
+
+    vents_coordinates = [
+        vent_coordinate.split("->") for vent_coordinate in vents_coordinates
+    ]
+    vents_coordinates = [
+        coordinate.strip()
+        for vent_coordinate in vents_coordinates
+        for coordinate in vent_coordinate
+    ]
+    vents_coordinates = [
+        vent_coordinate.split(",") for vent_coordinate in vents_coordinates
+    ]
+
+    return vents_coordinates
+
+
+def get_coordinate_pairs():
+    vents_coordinates = get_vents_input()
+    coordinate_pairs = []
+
+    biggest_x = 0
+    biggest_y = 0
+    j = 1
+    i = 0
+    for coordinates in vents_coordinates:
+        if j > len(vents_coordinates):
+            break
+        coordinate_pairs.append(
+            [
+                (int(vents_coordinates[i][0]), int(vents_coordinates[i][1])),
+                (int(vents_coordinates[j][0]), int(vents_coordinates[j][1])),
+            ]
+        )
+        i += 2
+        j += 2
+
+    for coordinates in coordinate_pairs:
+        for x, y in coordinates:
+            if x > biggest_x:
+                biggest_x = x
+            if y > biggest_y:
+                biggest_y = y
+
+    return coordinate_pairs, biggest_x, biggest_y
+
+
+def mark_coordinates_horizontal_vertical():
+    coordinate_pairs, biggest_x, biggest_y = get_coordinate_pairs()
+    coordinates_map = [[0 for i in range(biggest_x + 1)] for i in range(biggest_y + 1)]
+    x = 0
+    y = 1
+
+    for coordinates in coordinate_pairs:
```

Comment:
```
In [1]: a = [(("a", "b"), ("c", "d")), (("e", "f"), ("g", "h"))]

In [2]: for (x1, y1), (x2, y2) in a:
   ...:     print(x1, y1, x2, y2)
   ...:
a b c d
e f g h

In [3]:
```

Reply:
so i think i was trying to do something like that but because it had the 2 sets i was thinking you cant do it. this reads WAY better